docs/architecture/retry.md

```diff
@@ -0,0 +1,38 @@
+# Retry Mechanisms
+
+## gRPC
+
+### Interceptor
+
+All services use the gRPC interceptor `interceptor.RetryableInterceptor` that can retry a failed gRPC request.
+
+Its behavior is defined by:
+- `backoff.IsRetryable` which decides whether to retry based on the service error type
+- `backoff.RetryPolicy` which decides how long to backoff first - or not retry at all
+
+The `RetryableInterceptor` delegates the actual retry behavior to `backoff.ThrottleRetryContext` which:
+- never retries any context timeout/cancellation errors
+- retries `ResourceExhausted` service errors using a custom retry policy
+
+Tip: To inspect each service's retry behavior, look for its call to `NewRetryableInterceptor`.
```

Comment:
I'd rather keep this document as a launch pad to dive deeper - rather than attempting to detail the exact behavior only to see it go out of date sooner or later.